backend/services/theme_service.py

```
@@ -446,57 +446,37 @@ async def _generate_with_runware(
             try:
-                ip_weight_str = os.getenv("THEME_REFINE_IP_WEIGHT", "0.95") # FINAL FIX: Higher weight for face accuracy
+                ip_weight_str = os.getenv("THEME_REFINE_IP_WEIGHT", "0.85") # FINAL FIX v2: Lowered to prevent bleed
                 refinement_ip_weight = float(ip_weight_str)
                 # Clamp the value to a safe range (0.0 to 1.0)
                 if not (0.0 <= refinement_ip_weight <= 1.0):
                     logger.warning(f"⚠️ Invalid THEME_REFINE_IP_WEIGHT '{refinement_ip_weight}', clamping to range 0.0-1.0.")
                     refinement_ip_weight = max(0.0, min(1.0, refinement_ip_weight))
             except (ValueError, TypeError):
-                logger.warning(f"⚠️ Could not parse THEME_REFINE_IP_WEIGHT. Using default value 0.95.")
-                refinement_ip_weight = 0.95
+                logger.warning(f"⚠️ Could not parse THEME_REFINE_IP_WEIGHT. Using default value 0.85.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)